doc/importing-new-delivery-partner-data.md

```diff
@@ -0,0 +1,309 @@
+# Importing delivery partner data to the application
+
+## General
+We have enough confidence in our code to run this as a pair on production.
+
+If you want to gain confidence about any aspect and would get that by running
+the import locally first, feel free to do so. Running the import locally lets
+you get the data into perfect shape before you have to copy it over to
+production.
+
+All imports are run in a transaction, if any row fails the whole import is
+skipped.
+
+Once all the data is imported, zip up the actual files used and attach them to
+the Trello card (see below), this keeps a record of what might have been
+modified for the import and the original files.
+
+## Cards and source files
+We have a Trello card for each import to be run for each
+
+- delivery partner organisation
+- fund
+
+All of the files will be attached to the card:
+
+For Newton fund:
+
+- level B activity, actual spend and forecast data
+- level C activity, actual spend and forecast data
+- actual spend data pre formatted for import (no longer used and can be ignored)
+
+For GCRF:
+
+- level B activity, actual spend and forecast data
+- level C activity, actual spend and forecast data
+- level D activity, actual spend and forecast data
+- actual spend data pre formatted for import (no longer used and can be ignored)
+
+Pick up a card as a pair and move it to 'in progress'.
+
+The steps below should be followed in order as they are dependent on each other.
+
+## Activity
+
+Import activities in order:
+
+1. Level B
+2. Level C
+3. Level D
+
+### Checking activity data
+
+- are the headers correct? Use the migration template to verify.
+- are the values coded correctly? Use the migration template to verify.
+- are any values missing? The import will give you errors but look for anything
+  obvious.
+- raise any issues as soon as you can in Slack, the sooner we know the soon we
+  can work together to rectify them.
+
+### Prep activity data
+
+- correct any headers
+- for GCRF activities add a column `UK DP Named Contact` and set all rows to
+  `Must be provided`
+- add any missing values if possible - if not raise the issue in Slack
+- save as UTF8 csv file
+
+### Import activity data in production
+
+- copy the file over to production
+
+```
+cat FILENAME.csv | cf ssh beis-roda-prod -c "cat > FILENAME.csv"
+```
+
+- find the id of the `Organisation` that the import is for i.e. the Delivery Partner
+- connect to production:
+
+```
+cf login
+cf ssh beis-roda-prod
+```
+
+- run a rails console
+
+```
+cd /app && /usr/local/bin/ruby bin/rails console --sandbox
+```
+
+- get the id of the organisation you want
+
+```
+Organisation.all.pluck(:name, :id)
+```
+
+- quit the console
+
+- run the import 
+
+```
+cd /app && /usr/local/bin/ruby bin/rails activities:import CSV=FILENAME ORGANISATION_ID=ORGANISATION_ID
```

Comment:
We have now added the requirement to provide the email of the uploader to the task: https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/1058/commits/6af9d88a21bd4234ca894b5b3fe7a06c834f990c.

The uploader is then passed to the `ImportFromCsv` service and used to determine the permissions of the uploader via the organisation that they belong to, so you might want to make it clearer in the instructions that the uploader will have to change their account to belong to the correct organisation for the task they're running.